Horses_or_Humans - No Examples Yielded

**Short description**
Using Windows 10

Run:
dataset_name = "horses_or_humans"
dataset, info = tfds.load(name=dataset_name, split=tfds.Split.TRAIN, with_info=True)

Dataset will download, but when extracting will give error:

Shuffling and writing examples to C:\Users\lmoro\tensorflow_datasets\horses_or_humans\3.0.0.incomplete185N4P\horses_or_humans-train.tfrecord
Traceback (most recent call last):
  File "D:/bookwork/testwindows/venv/test.py", line 5, in <module>
    dataset, info = tfds.load(name=dataset_name, split=tfds.Split.TRAIN, with_info=True)
  File "D:\bookwork\testwindows\venv\lib\site-packages\wrapt\wrappers.py", line 567, in __call__
    args, kwargs)
  File "D:\bookwork\testwindows\venv\lib\site-packages\tensorflow_datasets\core\api_utils.py", line 52, in disallow_positional_args_dec
    return fn(*args, **kwargs)
  File "D:\bookwork\testwindows\venv\lib\site-packages\tensorflow_datasets\core\registered.py", line 305, in load
    dbuilder.download_and_prepare(**download_and_prepare_kwargs)
  File "D:\bookwork\testwindows\venv\lib\site-packages\wrapt\wrappers.py", line 606, in __call__
    args, kwargs)
  File "D:\bookwork\testwindows\venv\lib\site-packages\tensorflow_datasets\core\api_utils.py", line 52, in disallow_positional_args_dec
    return fn(*args, **kwargs)
  File "D:\bookwork\testwindows\venv\lib\site-packages\tensorflow_datasets\core\dataset_builder.py", line 334, in download_and_prepare
    download_config=download_config)
  File "D:\bookwork\testwindows\venv\lib\site-packages\tensorflow_datasets\core\dataset_builder.py", line 1030, in _download_and_prepare
    max_examples_per_split=download_config.max_examples_per_split,
  File "D:\bookwork\testwindows\venv\lib\site-packages\tensorflow_datasets\core\dataset_builder.py", line 883, in _download_and_prepare
    self._prepare_split(split_generator, **prepare_split_kwargs)
  File "D:\bookwork\testwindows\venv\lib\site-packages\tensorflow_datasets\core\dataset_builder.py", line 1058, in _prepare_split
    shard_lengths = writer.finalize()
  File "D:\bookwork\testwindows\venv\lib\site-packages\tensorflow_datasets\core\tfrecords_writer.py", line 211, in finalize
    self._shuffler.bucket_lengths, self._path)
  File "D:\bookwork\testwindows\venv\lib\site-packages\tensorflow_datasets\core\tfrecords_writer.py", line 88, in _get_shard_specs
    shard_boundaries = _get_shard_boundaries(num_examples, num_shards)
  File "D:\bookwork\testwindows\venv\lib\site-packages\tensorflow_datasets\core\tfrecords_writer.py", line 107, in _get_shard_boundaries
    raise AssertionError("No examples were yielded.")
AssertionError: No examples were yielded.


**Environment information**
* Operating System: Windows 10
* Python version: 3.7.3
* `tensorflow-datasets`/`tfds-nightly` version: 2.0.0
* `tensorflow`/`tensorflow-gpu`/`tf-nightly`/`tf-nightly-gpu` version: 2.0.0

**Reproduction instructions**

Run on Windows. The issue I believe is in the path separators. It's fine on Mac, Linux and Colab, but fails on Windows.

```
dataset_name = "horses_or_humans"
dataset, info = tfds.load(name=dataset_name, split=tfds.Split.TRAIN, with_info=True)
```

**Expected behavior**
TF Records to be extracted.

**Additional context**
When using 1.3.2, the .tfrecord files are extracted but empty. Upgrade to 2.0.0 gives this error, further upgrades (to 3.0.0) and the error stays.

--- a/tensorflow_datasets/image_classification/horses_or_humans.py
+++ b/tensorflow_datasets/image_classification/horses_or_humans.py
@@ -40,7 +40,7 @@ _TEST_URL = "https://storage.googleapis.com/laurencemoroney-blog.appspot.com/val
 _IMAGE_SIZE = 300
 _IMAGE_SHAPE = (_IMAGE_SIZE, _IMAGE_SIZE, 3)
 
-_NAME_RE = re.compile(r"^(humans|horses)/[\w-]*\.png$")
+_NAME_RE = re.compile(r"^(humans|horses)(/|\\)[\w-]*\.png$")
 
 
 class HorsesOrHumans(tfds.core.GeneratorBasedBuilder):

